src/codegen/events/solana.rs

```diff
@@ -21,6 +21,15 @@ pub(super) struct SolanaEventEmitter<'a> {
 }
 
 impl EventEmitter for SolanaEventEmitter<'_> {
+    fn selector(&self, _: usize) -> Vec<u8> {
+        let discriminator_image = format!("event:{}", self.ns.events[self.event_no].id);
+        let mut hasher = Sha256::new();
+        hasher.update(discriminator_image);
+        let result = hasher.finalize();
+
+        result[..8].to_vec()
```

Comment:
This is the same code we use for calculating the discriminator (on Solana is it not a selector) of a function. Please, use the same code we have there instead of re-writing it here.

Reply:
Good point, I've re-used the function `abi::anchor::discriminator`

Reply:
That does not work. event discriminators are hashed differently - the case is not touched (see anchor)

Reply:
There can be a common function to generate the discriminator image and hash it. We can continue fixing the casing for functions, but calling that common code afterwards.

Reply:
I've re-written it as you suggest.